src/main.ts

```diff
@@ -1,10 +1,34 @@
-export * from './components'
-export * from './compositions'
 export * from './errors'
 export * from './services'
 export * from './types'
+export type { RouterLinkProps, ToCallback } from '@/components/routerLink'
+export type { UseLink, UseLinkOptions } from './compositions/useLink'
+
+import { routerInjectionKey } from './keys'
+import { createRouterAssets } from './services/createRouterAssets'
+
+export const {
+  onBeforeRouteLeave,
+  onBeforeRouteUpdate,
+  onAfterRouteLeave,
+  onAfterRouteUpdate,
+  isRoute,
+  RouterView,
+  RouterLink,
+  useRoute,
+  useRouter,
+  useQueryValue,
+  useLink,
+} = createRouterAssets(routerInjectionKey)
 
 export type { CreateRouteOptions } from './types/createRouteOptions'
-export { isRoute } from './guards/routes'
 export { createRoute } from './services/createRoute'
 export { createExternalRoute } from './services/createExternalRoute'
+export { createRouterAssets } from './services/createRouterAssets'
+
+declare module 'vue' {
+  export interface GlobalComponents {
+    RouterView: typeof RouterView,
+    RouterLink: typeof RouterLink,
+  }
+}
```

Comment:
I think as some point we'll want to stop doing this automatically and include it in our global router installation just like the register interface.